pagopa/io-app
# feat: [IOBP-309, IOBP-312] Add error handling for payment verification and activation requests

## Short description
This PR adds the handling of failures during the verification and activation requests for a payment.

## List of changes proposed in this pull request
- Added `WalletPaymentFailure` type
- Added `WalletPaymentFailureDetail` component to display payment failures
- Added `usePaymentFailureSupportModal` hook to display the payment support request bottom sheet modal
- Added required locale keys

## How to test
Using the io-dev-api-server, checkout this branch https://github.com/pagopa/io-dev-api-server/pull/335 and add the following to the config.json:
```json
 "features": {
    "wallet": {
      "verificationFailure": {
        "faultCodeCategory": "PAYMENT_UNAVAILABLE",
        "faultCodeDetail":"PPT_PSP_SCONOSCIUTO"
      }
    }
  }
 ```
 With the IO app, navigate to the wallet payment playground and check that the failure is correctly displayed.

## Preview

https://github.com/pagopa/io-app/assets/6160324/03ec83b8-68c5-4d77-83a1-3bfd7dd8afcd
